File: src/Ddd.Tests/Trainings/Idea/App/TrainingIdeaService_edit_duration_test.cs
using Ddd.Tests.Trainings.Tests;
using Ddd.Trainings.Idea.Domain;
using Xunit;

namespace Ddd.Tests.Trainings.Idea.App;

public class TrainingIdeaService_edit_duration_test : TrainingTests
{
    [Fact]
    public void EditDuration_UpdatesDuration_OnValidInput()
    {
        // given
        var ideaId = TrainingIdeaBuilder.InDb().IdeaId;

        // when
        Service.EditDuration(ideaId.GetValue(), 3);

        // then
        var updatedIdea = IdeaRepo.FindById(ideaId);
        Assert.NotNull(updatedIdea);
        Assert.Equal(3, updatedIdea.Duration.Days);
    }

    [Fact]
    public void EditDuration_ThrowsException_WhenIdeaNotFound()
    {
        // expect
        var exception = Assert.Throws<Exception>(() => Service.EditDuration("non-existent-id", 3));
        Assert.Equal("Idea not found", exception.Message);
    }

    [Fact]
    public void EditDuration_ThrowsException_WhenDurationInvalid()
    {
        // given
        var ideaId = TrainingIdeaBuilder.InDb().GetId();

        // expect
        Assert.Throws<ArgumentException>(() => Service.EditDuration(ideaId.GetValue(), 0));
        Assert.Throws<ArgumentException>(() => Service.EditDuration(ideaId.GetValue(), 6));
        Assert.Throws<ArgumentException>(() => Service.EditDuration(ideaId.GetValue(), -1));
    }

    [Fact]
    public void EditDuration_ThrowsException_WhenIdeaIsProposed()
    {
        // given
        var idea = TrainingIdeaBuilder.Proposed(true).InDb();
        var ideaId = idea.GetId();

        // expect
        var exception = Assert.Throws<InvalidOperationException>(() => Service.EditDuration(ideaId.GetValue(), 3));
        Assert.Equal("Cannot edit duration of proposed idea", exception.Message);
    }
} 

File: src/test/java/com/ddd/trainings/idea/api/app/TrainingIdeaService_edit_duration_test.java
package com.ddd.trainings.idea.api.app;

import static org.assertj.core.api.Assertions.assertThat;
import static org.junit.jupiter.api.Assertions.assertThrows;

import com.ddd.trainings.TrainingTests;
import com.ddd.trainings.idea.domain.IdeaId;
import org.junit.jupiter.api.Test;

class TrainingIdeaService_edit_duration_test extends TrainingTests {

    @Test
    void editDuration_updatesDuration_onValidInput() {
        // given
        IdeaId ideaId = trainingIdeaBuilder.inDb().getId();

        // when
        service.editDuration(ideaId.getValue(), 3);

        // then
        var updatedIdea = ideaRepo.findById(ideaId).orElseThrow();
        assertThat(updatedIdea.getDuration().getDays()).isEqualTo(3);
    }

    @Test
    void editDuration_throwsException_whenIdeaNotFound() {
        // expect
        var exception = assertThrows(RuntimeException.class, 
            () -> service.editDuration("non-existent-id", 3));
        assertThat(exception.getMessage()).isEqualTo("Idea not found");
    }

    @Test
    void editDuration_throwsException_whenDurationInvalid() {
        // given
        IdeaId ideaId = trainingIdeaBuilder.inDb().getId();

        // expect
        assertThrows(IllegalArgumentException.class, 
            () -> service.editDuration(ideaId.getValue(), 0));
        assertThrows(IllegalArgumentException.class, 
            () -> service.editDuration(ideaId.getValue(), 6));
        assertThrows(IllegalArgumentException.class, 
            () -> service.editDuration(ideaId.getValue(), -1));
    }

    @Test
    void editDuration_throwsException_whenIdeaIsProposed() {
        // given
        var idea = trainingIdeaBuilder.proposed(true).inDb();
        var ideaId = idea.getId();

        // expect
        var exception = assertThrows(IllegalStateException.class, 
            () -> service.editDuration(ideaId.getValue(), 3));
        assertThat(exception.getMessage()).isEqualTo("Cannot edit duration of proposed idea");
    }
} 